### PR TITLE
Fix links and NeoForge mentions in README.md

### PR DESCRIPTION
**Pull request by Mishura4**
(Totally not an excuse to snatch PR #1)

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ NeoForge
 [![Discord](https://img.shields.io/discord/313125603924639766.svg?color=%237289da&label=Discord&logo=discord&logoColor=%237289da)][Discord] 
 
 
-Forge is a free, open-source modding API all of your favourite mods use!
+NeoForge is a free, open-source modding API all of your favourite mods use!
```
Comment (embeddedt) [inline docs/README.md]: Third time's the charm?

```suggestion
NeoForge is a free, open-source modding API many of your favourite mods use!
```

**Review by Su5eD — APPROVED** [review]

**Comment by cpw** [comment]
We'll have to see if this still works after the big update that is currently sitting on my computer, heh.

**Review by SizableShrimp — CHANGES_REQUESTED** [review]
I went ahead and rebased this PR on the newest 1.20.x changed git history for the benefit of the PR author.
Now for my thoughts:
If we're updating the README, might as well do it right.

> Installing NeoForged

NeoForged is the organization, not the product.
Additionally, this section is bad in general. It links to a dead placeholder curseforge link, which can be confusing to users. And, the information about Latest builds being more unstable is basically bunk nowadays. We should probably delete this section while we figure out what our installation will look like.

> NeoForged is a large project with many collaborators working on it around the clock.

Same issue, it's NeoForge, not NeoForged.

The CLA link and CONTRIBUTING.md doc are both outdated, but those should probably be saved for changing on another day.

**Comment by hypersmc** [comment]
Please update the download button from the link: https://maven.neoforged.net/releases/net/neoforged/forge/
to: https://maven.neoforged.net/#/releases/net/neoforged/forge/

**Comment by cpw** [comment]
Updated to latest tip so this should run all the checks properly and then be ready to merge.

**Comment by Mishura4** [comment]
Changed some mentions of NeoForged to NeoForge and vice versa as requested (see https://github.com/neoforged/NeoForge/pull/1#pullrequestreview-1527768434)

Also updated some docs links that were now dead

File: docs/README.md [hunk 2]
```diff
@@ -29,40 +29,40 @@ For support and questions, visit [the NeoForged Discord server][Discord].
...
 [Documentation]: https://docs.neoforged.net/
-[Getting-Started]: https://docs.neoforged.net/en/latest/gettingstarted/
-[ForgeDev]: https://docs.neoforged.net/en/latest/forgedev/
-[Pull-Requests]: https://docs.neoforged.net/en/latest/forgedev/#making-changes-and-pull-requests
-[CurseForge]: https://curseforge.com/placeholder
\ No newline at end of file
+[Getting-Started]: https://docs.neoforged.net/docs/gettingstarted/
+[NeoForgeDev]: https://docs.neoforged.net/docs/forgedev/
+[Pull-Requests]: https://docs.neoforged.net/docs/forgedev/#making-changes-and-pull-requests
+[CurseForge]: https://curseforge.com/placeholder
```
Comment (DataDalton) [inline docs/README.md:68]: Should probably remove this weird placeholder link that's broken and just link directly to Curseforge, same for main README

File: docs/README.md [hunk 3]
```diff
@@ -29,40 +29,40 @@ For support and questions, visit [the NeoForged Discord server][Discord].
```
Comment (DataDalton) [inline docs/README.md:29]: Should probably rewrite this original installing NeoForge thing as it doesn't make sense and should say how to install NeoForge without going through Curseforge and linking to the installer on https://neoforged.net/

**Comment by Technici4n** [comment]
See #232.

**Pull request closed without merging**